llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7772,12 +7551,23 @@ VectorizationFactor LoopVectorizationPlanner::computeBestVF() {
 
       InstructionCost Cost = cost(*P, VF);
       VectorizationFactor CurrentFactor(VF, Cost, ScalarCost);
-      if (isMoreProfitable(CurrentFactor, BestFactor))
-        BestFactor = CurrentFactor;
-
       // If profitable add it to ProfitableVF list.
       if (isMoreProfitable(CurrentFactor, ScalarFactor))
         ProfitableVFs.push_back(CurrentFactor);
```

Comment:
Previously when the VFs were pruned in getMaximizedVFForTarget I presume they wouldn't have been added to the ProfitableVFs list either. Should the register usage check be moved above this too?

Reply:
Thanks for spotting that, done.